core/base/src/TString.cxx

```diff
@@ -361,7 +382,11 @@ TString& TString::operator=(const TSubString &substr)
 
 TString& TString::Append(char c, Ssiz_t rep)
 {
-   if (!rep) return *this;
+   if (rep <= 0 ){
+      Error("TString& TString::Append", "Negative length!");
```

Comment:
That's not necessarily true, `rep` could be 0.

Reply:
No, no, the `if` clause was correct (compare the existing code).  But the error should only be raised for truly negative values of `rep`.